integrations/clickup/client.py

```diff
@@ -0,0 +1,112 @@
+from typing import Any, AsyncGenerator
+from httpx import Timeout
+from loguru import logger
+from port_ocean.utils import http_async_client
+
+PAGE_SIZE = 50
+
+
+class ClickUpClient:
+    def __init__(self, clickup_token: str) -> None:
+        self.headers = {"Authorization": clickup_token}
+
+        self.client = http_async_client
+        self.client.headers.update(self.headers)
+        self.client.timeout = Timeout(30)
+
+    @staticmethod
+    def _generate_base_req_params(
+        max_results: int = PAGE_SIZE, start_at: int = 0
+    ) -> dict[str, Any]:
+        return {
+            "page": start_at // max_results,
+            "limit": max_results,
+        }
+
+    async def _get_paginated_teams(
+        self, host: str, params: dict[str, Any]
+    ) -> dict[str, Any]:
+        response = await self.client.get(f"{host}/api/v2/team", params=params)
+        response.raise_for_status()
+        return response.json()
+
+    async def _get_paginated_spaces(
+        self, host: str, team_id: str, params: dict[str, Any]
+    ) -> dict[str, Any]:
+        response = await self.client.get(
+            f"{host}/api/v2/team/{team_id}/space", params=params
+        )
+        response.raise_for_status()
+        return response.json()
+
+    async def _get_paginated_projects(
+        self, host: str, space_id: str, params: dict[str, Any]
+    ) -> dict[str, Any]:
+        response = await self.client.get(
+            f"{host}/api/v2/space/{space_id}/list", params=params
+        )
+        response.raise_for_status()
+        return response.json()
+
```

Comment:
Please rename the methods accordingly. They are all not doing any pagination so rename them to get_teams(), get_projects() etc